geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -607,6 +669,87 @@ private void addIfMatching(GlobPattern matchPattern, List<byte[]> resultList, by
     }
   }
 
+  private void computeIntersection(List<RedisSortedSet> sets, ZAggregator aggregator) {
+    RedisSortedSet retVal = new RedisSortedSet(Collections.emptyList(), new double[] {});
+    RedisSortedSet smallestSet = sets.get(0);
+
+    for (RedisSortedSet set : sets) {
+      if (set.getSortedSetSize() < smallestSet.getSortedSetSize()) {
+        smallestSet = set;
+      }
+    }
+
+    for (byte[] member : smallestSet.members.keySet()) {
+      Double newScore;
+      if (aggregator.equals(ZAggregator.SUM)) {
+        newScore = getSumOfScoresForMember(sets, member);
+      } else if (aggregator.equals(ZAggregator.MAX)) {
+        newScore = getMaxScoreForMember(sets, member);
+      } else {
+        newScore = getMinScoreForMember(sets, member);
+      }
+
+      if (newScore != null) {
+        if (newScore.isNaN()) {
```

Comment:
the new "zinterstore" impl does not do this NaN check. Did you decide it was not needed?

Reply:
Yes, I believe it is now unnecessary because the ZAggregator function that is applied doesn't ever return null.